com.unity.render-pipelines.core/Runtime/Textures/MSAASamples.cs

```diff
@@ -14,4 +18,52 @@ public enum MSAASamples
         /// <summary>MSAA 8X.</summary>
         MSAA8x = 8
     }
+
+    public static class MSAASamplesExtensions
+    {
+        public static FormatUsage AsFormatUsage(this MSAASamples samples)
+        {
+            switch (samples)
+            {
+                case MSAASamples.MSAA2x:
+                    return FormatUsage.MSAA2x;
+                case MSAASamples.MSAA4x:
+                    return FormatUsage.MSAA4x;
+                case MSAASamples.MSAA8x:
+                    return FormatUsage.MSAA8x;
+                default:
+                    return FormatUsage.Render;
+            }
+        }
+
+        public static MSAASamples Validate(this MSAASamples samples)
+        {
+            if (samples == MSAASamples.None)
+                return samples;
+
+            GraphicsFormat ldrFormat = SystemInfo.GetGraphicsFormat(DefaultFormat.LDR);
```

Comment:
Why do you use SystemInfo.GetGraphicsFormat(DefaultFormat.LDR); ?  to check this.
We need to check every format that will use MSAA.
i.e ColorBufferFormat (which can be selected to be RGBA8 and RGBA16F) for example and other RT like R32 we use for depth in prepass when MSAA is enabled and other stuff. We don't have a single format here.

Is the intent of this function to only check if RGBA8 MSAA sample count is supported or I miss something?

Reply:
The idea is to use any texture format to figure out what is the generic supported range of MSAA samples on your GfxDevice, it's an improvement over to current assumption.

I don't think our C# APIs have any better way to query a supported MSAA range in other way (if your GfxDevice doesn't support 8x but 2/4 only, or if some really old GPUs wouldn't support 2x). Shaders could pick up some 8x variant for MSTextures use, so this isn't something that we can silently fallback underneath SRP within GfxDevice implementation.